tests/lib/rules/no-container.test.ts

```diff
@@ -0,0 +1,82 @@
+import { createRuleTester } from '../test-utils';
+import rule, { RULE_NAME } from '../../../lib/rules/no-container';
+
+const ruleTester = createRuleTester({
+  ecmaFeatures: {
+    jsx: true,
+  },
+});
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
```

Comment:
Could you add an additional valid test for using `container` not related to Testing Library? Something like:

```javascript
function getExampleDOM() {
  const container = document.createElement('div')
  container.innerHTML = `
    <label for="username">Username</label>
    <input id="username" />
    <button>Print Username</button>
  `
  const button = container.querySelector('button')

  button.addEventListener('click', () => console.log('clicked'));
  return container;
}

const exampleDOM = getExampleDOM();

screen.getByText(exampleDOM, 'Print Username').click();
```

This is a simplified code from [DOM Testing Library example](https://testing-library.com/docs/dom-testing-library/example-intro).